src/components/paradox/results/personality/career/inline-jobs/auto.js

```diff
@@ -0,0 +1,59 @@
+import PropTypes from "prop-types";
+import {useEffect, useRef, useState} from "react";
+import InlineJobs from "./index";
+import style from "./style.scss";
+
+function InlineJobsAuto({jobs, jobSource, translate}) {
+  const ref = useRef(null);
+  const [count, setCount] = useState(0);
+
+  /** Adjusts number of jobs based on screen size */
+  const configureCount = () => {
+    const INLINE_JOB_WIDTH = 180;
+    const NEXT_BUTTON_SIZE = 44;
+    const {offsetLeft: OFFSET_LEFT, offsetWidth: WIDTH} = ref.current;
+    const {offsetWidth: PARENT_WIDTH} = ref.current.offsetParent;
+    const WIDTH_ESTIMATE = PARENT_WIDTH - OFFSET_LEFT;
+    const DIFFERENCE = WIDTH_ESTIMATE - WIDTH;
+
+    const VALUE = DIFFERENCE <= 0 ? Math.floor(PARENT_WIDTH - OFFSET_LEFT) : WIDTH;
+
+    const newCount = Math.floor((VALUE - (NEXT_BUTTON_SIZE * 2)) / INLINE_JOB_WIDTH);
+    setCount(newCount);
+  };
+
+  useEffect(() => {
```

Comment:
This may also need to be a useLayoutEffect if it depends on the ref, but I'm not sure about that

Reply:
This is a good catch! It Since the function calculates and updates the count, it should run before the component is visually updated.